### PR TITLE
T527 collections page

### LINKED ISSUE #533: Files without thumbnails result in broken links
Bug discovery on the test and production servers. 🐛 

Files without thumbnails are redirected to `images/default.png`, a path that does not exist. We will need `assets/default-[generated hash].png`, but more research is necessary regarding how to generate this here.

Failed ideas have been to replace the definition of `default_image` in `app/services/hyrax/thumbnail_path_service.rb` with something like a `Rails.root.join("app", "assets", "images", "default.png")` (this did nothing and the view still generated `images/default.png`).

Also found [this Hyrax issue](https://github.com/samvera/hyrax/issues/1095), which seems to have been closed years ago without being fixed. 
If fixing this is impractical, altering the messy display of the alt text would be a good fallback.

screenshot displaying bug
HTML element inspected

### PR DESCRIPTION
**Pull request by maxturer**
This will close #533 and #527.
Includes: 

1. New collections page involving view, route, controller, and presenter referencing Hyrax Collection entity
2. Updated default icons for collections, single items, and audio files.
3. Bug fix for descriptions not displaying under collections
4. Bug fix for margins not displaying on 404 pages
5. Bug fix for dividers between navigation menu items not displaying on search pages

### CODE REVIEW

File: app/controllers/hyrax/collections_page_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    class_attribute :presenter_class
+    self.presenter_class = Hyrax::CollectionsPagePresenter
+
+    def index
+        @presenter = presenter_class.new(collections)
+    end
+
+    private
+
+    # Return 5 collections
```
Comment (dolsysmith) [inline app/controllers/hyrax/collections_page_controller.rb:16]: I'm assuming this comment is leftover from the home page collections controller -- the collections page seems to display as many collections as are visible to the user, not just 5.

**Review by dolsysmith — CHANGES_REQUESTED** [review]
- The new icons look great!
- The collections page has a nice, clean look. One tiny suggestion: should `Browse all Collections in GW ScholarSpace` say `Browsing` instead? The imperative mood made me think it was a button at first.
- I added collections of various types of visibility, and I can confirm that the private and authenticated collections are not visible to non-authenticated users, and that private works are not visible to authenticated users who did not create them.

**A bug**
- I'm not seeing the collection descriptions appear. To fix the problem, line 6 in `app/models/collection.rb` needs to be uncommented. A reindexing is also required (for existing collections with descriptions.)

**A question**
I notice that a collection can be assigned the thumbnail of one of its works, and this thumbnail will then appear beside the collection in the search and on the collections page. But a collection can also have a logo (set under the Branding tab); this will appear beside the collection name on the page for that collection, but it will _not_ appear in the search results or on the collections index page. That behavior is maybe a little counterintuitive to me, but perhaps it won't be to others.

File: app/controllers/hyrax/collections_page_controller.rb (hunk 1 above)
Comment (maxturer) [reply app/controllers/hyrax/collections_page_controller.rb]: That was me troubleshooting, you're right. Removed!

**Review by dolsysmith — APPROVED** [review]
Looks good to me!

**Pull request merged**